test/cpp/end2end/end2end_test.cc

```diff
@@ -405,6 +410,7 @@ class End2endTest : public ::testing::TestWithParam<TestScenario> {
 
     stub_ = grpc::testing::EchoTestService::NewStub(channel_);
     DummyInterceptor::Reset();
+    WaitOnServerAuthorizationToComplete(GetCredentialsProvider());
```

Comment:
It seems `ResetStub` API will be triggered for non-TLS cases as well in which case we do not need server authorization.  Is there a way to avoid `WaitOnServerAuthorizationToComplete` call on the path common to all credentials?

Reply:
I added a `credentials_type` parameter to `WaitOnSpawnedThreads`. If `credentials_type` is anything other than `kSpiffeCredentialsType`, then it immediately returns.